baseapp/baseapp_test.go

```diff
@@ -161,9 +162,17 @@ func setupBaseAppWithSnapshots(t *testing.T, blocks uint, blockTxs int, options
 		app.EndBlock(abci.RequestEndBlock{Height: height})
 		app.Commit()
 
-		// Wait for snapshot to be taken, since it happens asynchronously. This
-		// heuristic is likely to be flaky on low-IO machines.
-		time.Sleep(time.Duration(int(height)*blockTxs) * 200 * time.Millisecond)
+		// Wait for snapshot to be taken, since it happens asynchronously.
+		if uint64(height)%snapshotInterval == 0 {
+			for {
+				snapshot, err := snapshotStore.Get(uint64(height), snapshottypes.CurrentFormat)
+				require.NoError(t, err)
+				if snapshot != nil {
+					break
+				}
+				time.Sleep(100 * time.Millisecond)
+			}
+		}
```

Comment:
nit: I'm sure this is always guaranteed to complete as much as you can guarantee anything in a test, but would it be safer to use a ticker/timeout select combo?

Reply:
Was thinking about it, but the test timeout will trigger after like 2 minutes anyway.

Reply:
Sure. Let's just hope no one runs a timeout of like 15m haha

Reply:
Sure, added a timeout just in case.